subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -167,25 +149,37 @@ int aws_fota_parse_update_job_exec_state_rsp(char *update_rsp_document,
 	return ret;
 }
 
-int aws_fota_parse_notify_next_document(char *job_document,
-		u32_t payload_len, char *job_id_buf, char *hostname_buf,
-		char *file_path_buf)
+int aws_fota_parse_DescribeJobExecution_rsp(char *job_document,
+					   u32_t payload_len,
+					   char *job_id_buf,
+					   char *hostname_buf,
+					   char *file_path_buf,
+					   int *version_number)
 {
-	struct notify_next_obj job;
+	struct desc_job_execution_obj root_obj;
 	struct job_document_obj *job_doc_obj;
+	struct execution_obj *exec_obj;
 
-	int ret = json_obj_parse(job_document,
-				 payload_len,
-				 notify_next_obj_descr,
-				 ARRAY_SIZE(notify_next_obj_descr),
-				 &job);
-	job_doc_obj = &job.execution.job_document;
+	int ret = json_obj_parse(job_document, payload_len,
+				 desc_job_execution_obj_descr,
+				 ARRAY_SIZE(desc_job_execution_obj_descr),
+				 &root_obj);
+
+	if (!(ret & BIT(2))) {
+		/* No execution object in JSON */
+		return 1;
+	}
 
 	/* Check if the execution field of the object has been parsed */
-	if (ret & 0x02) {
-		if (job.execution.job_id != 0) {
-			strncpy_nullterm(job_id_buf, job.execution.job_id,
-				      AWS_JOBS_JOB_ID_MAX_LEN);
+	if (ret & BIT(2)) {
```

Comment:
if this check necessary?  `if (!(ret & BIT(2)))` returns above

Reply:
I think you are correct. It shouldn't be necessary to check if the execution object was successfully decoded as we checks if it wasn't and returns. Thanks for spotting this :)